Implements GitHub Actions

helrond opened the pull request:
- Adds deploy and branch enforcement workflows. 
- Updates dependency workflow
- Removes Travis CI configs and build status from README

**NOTEs**:
- No test workflow was added since there are no unit tests for this repo
- A step to copy the theme config file was added to the deploy workflow
- The `BASENAME` variable does not appear to be needed in the deploy workflow, so I removed it.

helrond committed:
implement GHA

HaSistrunk commented on .github/workflows/dependencies.yml:
```diff
@@ -14,9 +14,17 @@ jobs:
+          persist-credentials: false
 
       - name: Set up Python
         uses: actions/setup-python@v4
```
```suggestion
        uses: actions/setup-python@v5
```

HaSistrunk reviewed (commented):
The styles deploy has an action to [invalidate the cloudfront cache](https://github.com/RockefellerArchiveCenter/styles/blob/861be22d0a0dbd3528a229fa4c5283c7d4f35f5f/.github/workflows/deploy_styleguide.yml#L57). Is that something we can do in this repo to address issue #185 ?

helrond committed:
Update setup-python version

Co-authored-by: Hannah Sistrunk <hsistrunk@gmail.com>

helrond commented:
> The styles deploy has an action to [invalidate the cloudfront cache](https://github.com/RockefellerArchiveCenter/styles/blob/861be22d0a0dbd3528a229fa4c5283c7d4f35f5f/.github/workflows/deploy_styleguide.yml#L57). Is that something we can do in this repo to address issue #185 ?

Great call!

@p-galligan does the IAM user/role associated with this repo have permissions to create Cloudfront cache invalidations?

p-galligan commented:
> > The styles deploy has an action to [invalidate the cloudfront cache](https://github.com/RockefellerArchiveCenter/styles/blob/861be22d0a0dbd3528a229fa4c5283c7d4f35f5f/.github/workflows/deploy_styleguide.yml#L57). Is that something we can do in this repo to address issue #185 ?
> 
> Great call!
> 
> @p-galligan does the IAM user/role associated with this repo have permissions to create Cloudfront cache invalidations?

It did not, but I have changed up the role and added permissions to invalidate cloudfront cache now.

p-galligan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,73 @@
+          aws ecr put-image --registry-id ${{ secrets.AWS_ACCOUNT_ID }} --repository-name ${{ env.APP_NAME }} --image-tag prod --image-manifest "$MANIFEST"
+      - name: Deploy lambda
+        run: aws lambda update-function-code
+          --function-name arn:aws:lambda:$AWS_REGION:$AWS_LAMBDA_ACCOUNT_ID:function:$LAMBDA_NAME
```
Actually just found two issues with this.

We need an if statement to check for the base (since we only want to deploy from prod images).

And `$AWS_LAMBDA_ACCOUNT_ID` should actually be `${{ secrets.AWS_ACCOUNT_ID }}`

helrond committed:
correct variable and add conditional

helrond commented:
OK I added the conditional and changed the variable to the correct value. In looking at the Cloudfront invalidation - I realized this deploy is not actually building the site, so adding it to the deploy here won't do anything. Instead, that logic needs to be added to `update.py`.
